pkg/core/mempool/mem_pool.go

```diff
@@ -240,59 +253,109 @@ func (mp *Pool) Add(t *transaction.Transaction, fee Feer, data ...interface{}) e
 	return nil
 }
 
-// Remove removes an item from the mempool, if it exists there (and does
-// nothing if it doesn't).
-func (mp *Pool) Remove(hash common.Hash, feer Feer) {
+// get pending nonce of from
+func (mp *Pool) PendingNonce(addr common.Address) uint64 {
 	mp.lock.Lock()
-	mp.removeInternal(hash, feer)
-	mp.lock.Unlock()
+	defer mp.lock.Unlock()
+	return mp.pendingNonces[addr]
 }
 
-// removeInternal is an internal unlocked representation of Remove.
-func (mp *Pool) removeInternal(hash common.Hash, feer Feer) {
-	if tx, ok := mp.verifiedMap[hash]; ok {
-		var num int
-		delete(mp.verifiedMap, hash)
-		for num = range mp.verifiedTxes {
-			if hash == (mp.verifiedTxes[num].txn.Hash()) {
-				break
-			}
+// check the current tx is continue in memory pool
+func (mp *Pool) CheckNonceContinue(tx *transaction.Transaction, dbNonce uint64) bool {
```

Comment:
I'm not sure we need this check, considering we sort txes by nonce in mempool already.

Reply:
When the policy fee increase, we need remove the tx with low gas and the nonce after it with same sender